doc/man1/flux-uri.rst

```diff
@@ -0,0 +1,144 @@
+URIs indicate the "connector" which will be used to establish the
+connection, and are typically either *local*, with a  ``local`` URI
+scheme, or *remote*, with a ``ssh`` URI scheme. These URIs are considered
+fully-resolved, native Flux URIs.
+
+Processes running within a Flux instance will have the ``FLUX_URI``
+environment variable set to a native URI which :man3:`flux_open` will
+use by default. Therefore, there is usually no need to specify a URI when
+connecting to the enclosing instance. However, connecting to a _different_
+Flux instance will require discovery of the fully-resolved URI for that
+instance.
+
+**flux uri** attempts to resolve its *TARGET* argument to a native local
+or remote URI. The *TARGET* is itself a URI which specifies the method
+to use in URI resolution via the scheme part, with the path and query
+parts passed to a plugin which implements the resolution method.
+
+As a convenience, if *TARGET* is specified with no scheme, then the scheme
+is assumed to be ``jobid``.  This allows ``flux uri`` to be used to look
+up the URI for a Flux instance running as a job with:
+
+::
+
+   $ flux uri JOBID
+
+Depending on the *TARGET* URI scheme and corresponding plugin, specific
+query arguments int *TARGET* may be supported. However, as a convenience,
+all *TARGET* URIs support the special query arguments ``local`` or
+``remote`` to force the resulting URI into a local (``local://``) or remote
+(``ssh://``) form. For example:
+
+::
+
+   $ flux uri JOBID?local
+
+would return the ``local://`` URI for *JOBID* (if the URI can be resolved).
+
+A list of supported URI schemes will be listed at the bottom of
+``flux uri --help`` message. For a description of the URI resolver schemes
+included with Flux, see the URI SCHEMES and EXAMPLES sections below.
+
+OPTIONS
+=======
+
+**-remote**
+   Return the _remote_ (``ssh://``)  equivalent of the resolved URI.
+
+**--local**
+   Return the _local_ (``local://``) equivalent of the resulved URI.
+   Warning: the resulting URI may be invalid for the current system
+   if the network host specified by an ``ssh`` URI is not the current
+   host.
+
+URI SCHEMES
+===========
+
+The following URI schemes are included by default:
+
+jobid:ID[/ID...]
+   This scheme attempts to get the URI for a Flux instance running as a
+   job in the current enclosing instance. This is the assumed scheme if no
+   ``scheme:`` is provided in *TARGET* passed to ``flux uri``, so the
+   ``jobid:`` prefix is optional. A hierarchy of Flux jobids is supported,
+   so ``f1234/f3456`` will resolve the URI for job ``f3456`` running in
+   job ``f1234`` in the current instance.
+
+pid:PID
+  This scheme attempts to read the ``FLUX_URI`` value from the process id
+  *PID* using ``/proc/PID/environ``. If *PID* refers to a ``flux-broker``,
+  then a child of the broker is used instead in order to obtain the
+  URI for that broker (instead of its parent)
```

Comment:
Maybe instead of "then a child of the broker is used instead" say "then the scheme reads FLUX_URI from the broker's initial program or another child process since FLUX_URI in the broker's environment would refer to _its_ parent.

I'm not sure if that makes it any clearer.  Feel free to ignore.

Missing period at end of sentence.

Reply:
Also added "(or may not be set at all in the case of an instance started with `flux start --test-size=N`)"